sw/applications/src/utils/utils.h

```diff
@@ -0,0 +1,44 @@
+// Copyright 2020 ETH Zurich and University of Bologna.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "layer.h"
+#include "snrt.h"
+
+/**
+ * @brief returns cycle number and injects maker
+ * to track performance
+ *
+ * @return uint32_t
+ */
+uint32_t benchmark_get_cycle();
+
+/**
+ * @brief start tracking of dma performance region
+ *
+ */
```

Comment:
Perhaps mention that this has no HW side-effects, just for tracking in trace files. Same of course for `stop`